cmd/server/assets/users/import.html

```diff
@@ -22,19 +23,134 @@ <h1>Import users</h1>
     <p>Bulk import a list of users to <strong>{{.currentRealm.Name}}</strong>.</p>
 
     <div class="card mb-3 shadow-sm">
-      <div class="card-header">Import options</div>
+      <div class="card-header">Import</div>
       <div class="card-body">
         <div class="alert alert-warning">
           <span class="oi oi-beaker"></span> This feature is still under
           active development.
         </div>
-        <small class="form-text text-muted">Coming soon</small>
+
+        <form id="form">
+          <div class="custom-file">
+            <input type="file" class="custom-file-input" id="csv" accept=".csv" required>
+            <label class="custom-file-label" for="csv" id="fileLabel">Select a CSV file</label>
+          </div>
+          <button class="btn btn-primary" type="submit" id="import" disabled>Import users</button>
+        </form>
+      </div>
+
+      <div class="card-body">
+        <div class="progress">
+          <div id="progress" class="progress-bar" role="progressbar" aria-valuenow="0" aria-valuemin="0"
+            aria-valuemax="100"></div>
+        </div>
+        <table class="table table-bordered" id="csvtable" style="display:none;">
+          <thead>
+            <tr>
+              <th>Email</th>
+              <th>Name</th>
+            </tr>
+          </thead>
+          <tbody id="csvtablebody"></tbody>
+        </table>
       </div>
     </div>
-    <a class="card-link" href="/users">&larr; All users</a>
+    <a class=" card-link" href="/users">&larr; All users</a>
   </main>
 
   {{template "scripts" .}}
+  <script type="text/javascript">
```

Comment:
Hmmm this is a lot of javascript. Is there a strong reason to do this on the client vs the server? It seems like a traditional form submit + background processing would suffice here.

Reply:
I'm also worried about the "modern" requirements here. When you think about most hospitals and medical facilities, they aren't running the latest and greatest browsers. I wouldn't be surprised if we have some clients using IE on XP still, with no support for the HTML5 file objects.

Reply:
The main reason is so that it can batch and update progress as it goes. If we submit the whole file, we still need to return a response with all the users so we can do the password reset in javascript.